src/Microsoft.OpenApi/Helpers/DictionaryCloneHelper.cs

```diff
@@ -18,9 +18,9 @@ public class DictionaryCloneHelper
         /// <typeparam name="U">The type of the value of the dictionary.</typeparam>
         /// <param name="dictionary">The target dictionary to clone.</param>
         /// <returns>The cloned dictionary.</returns>
-        public static Dictionary<T, U> Clone<T, U>(IDictionary<T, U> dictionary)
-        {
-            var clonedDictionary = new Dictionary<T, U>();
+        internal static Dictionary<T, U> Clone<T, U>(IDictionary<T, U> dictionary)
+        {            
+            var clonedDictionary = dictionary is null ? null : new Dictionary<T, U>();
```

Comment:
```suggestion
            if (dictionary is null) return null;
            var clonedDictionary = new Dictionary<T, U>();
```

This way you can remove the condition below